Source/Scene/UrlTemplateImageryProvider.js

```diff
@@ -113,15 +113,22 @@ define([
      *                  be treated as if their alpha is 1.0 everywhere.  When this property is false, memory usage
      *                  and texture upload time are potentially reduced.
      * @param {GetFeatureInfoFormat[]} [options.getFeatureInfoFormats] The formats in which to get feature information at a
-     *                                 specific location when {@see UrlTemplateImageryProvider#pickFeatures} is invoked.  If this
+     *                                 specific location when {@link UrlTemplateImageryProvider#pickFeatures} is invoked.  If this
      *                                 parameter is not specified, feature picking is disabled.
-     *
+     * @param {Boolean} [options.deferReadiness=false] True to defer full construction of this instance until a later call to {@link UrlTemplateImageryProvider#reinitialize}.
```

Comment:
I know I'm late to the party, but I think the approach here is confusing and neither `deferReadiness` or `hasReadyPromise` are needed.  I feel there are two cleaner options:

1. Have `createTileMapServiceImageryProvider` just return a Promise that resolves to the provider when everything is initialized. The main reason against this would be because users always struggle with promises and we want all of the `createXXXImageryProvider` style functions to return `ImageryProvider` instances to keep the api simpler.

2. Have UrlTermplateProvider allow `options` to also be a promise to options, which when resolved would configure the provider and set ready.  No need to manually call initialize and if options is not a promise the code is exactly the same:

```javascript 
when(options).then(function(){ 
    // constructor code.  Options can be either a promise or an object.
});
```

After thinking about it some, I think 2 is the right way to go.